File: src/main/kotlin/com/github/itsempa/cyclamax/config/categories/MiscConfig.java
package com.github.itsempa.cyclamax.config.categories;

import com.google.gson.annotations.Expose;
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean;
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorSlider;
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption;

public class MiscConfig {

    @Expose
    @ConfigOption(name = "Auto Updates", desc = "Automatically notify of new updates.")
    @ConfigEditorBoolean
    public boolean autoUpdates = true;

    @Expose
    @ConfigOption(name = "Spin", desc = "weeeeeeee")
    @ConfigEditorBoolean
    public boolean spin = false;

    @Expose
    @ConfigOption(name = "Spin Speed", desc = "How fast you spin")
    @ConfigEditorSlider(minValue = -100, minStep = 1, maxValue = 100)
    public int spinSpeed = 50;

}

File: src/main/kotlin/com/github/itsempa/cyclamax/config/categories/MiscConfig.kt
package com.github.itsempa.cyclamax.config.categories

import at.hannibal2.skyhanni.deps.moulconfig.annotations.ConfigEditorBoolean
import at.hannibal2.skyhanni.deps.moulconfig.annotations.ConfigEditorSlider
import at.hannibal2.skyhanni.deps.moulconfig.annotations.ConfigOption
import com.google.gson.annotations.Expose

class MiscConfig {
    @Expose
    @ConfigOption(name = "Spin", desc = "weeeeeeee")
    @ConfigEditorBoolean
    var spin: Boolean = false

    @Expose
    @ConfigOption(name = "Spin Speed", desc = "How fast you spin")
    @ConfigEditorSlider(minValue = -100f, minStep = 1f, maxValue = 100f)
    var spinSpeed: Int = 50

    @Expose
    @ConfigOption(name = "Stop Sword Blocking", desc = "makes it so that you dont get the block sword animation")
    @ConfigEditorBoolean
    var stopSwordBlocking: Boolean = false

    @Expose
    @ConfigOption(
        name = "Remove Armor Enchant Glint",
        desc = "removes enchant glint only from armor (worn or in item form)"
    )
    @ConfigEditorBoolean
    var removeArmorGlint: Boolean = false
}
